### ISSUE TITLE
SCSSLint::Linter::ColorVariable raised unexpected error linting file sass/comments2.scss: 'undefined method `[]' for nil:NilClass

### ISSUE THREAD

**Issue by bassjobsen** [brigade/scss-lint#464]
code:

```css
$a: 0;

/*!
* test #{a}.
*/
```

    SCSSLint::Linter::ColorVariable raised unexpected error linting file sass/comments2.scss: 'undefined method `[]' for nil:NilClass'
    /var/lib/gems/1.9.1/gems/scss_lint-0.38.0/lib/scss_lint/linter/color_variable.rb:40:in `literal_string?'
    /var/lib/gems/1.9.1/gems/scss_lint-0.38.0/lib/scss_lint/linter/color_variable.rb:18:in `visit_script_string'
    /var/lib/gems/1.9.1/gems/sass-3.4.13/lib/sass/tree/visitors/base.rb:36:in `visit'
    /var/lib/gems/1.9.1/gems/scss_lint-0.38.0/lib/scss_lint/linter.rb:131:in `visit'
    /var/lib/gems/1.9.1/gems/scss_lint-0.38.0/lib/scss_lint/linter.rb:142:in `block in visit_children'
    /var/lib/gems/1.9.1/gems/scss_lint-0.38.0/lib/scss_lint/linter.rb:140:in `each'
    /var/lib/gems/1.9.1/gems/scss_lint-0.38.0/lib/scss_lint/linter.rb:140:in `visit_children'
    /var/lib/gems/1.9.1/gems/sass-3.4.13/lib/sass/tree/visitors/base.rb:38:in `visit'
    /var/lib/gems/1.9.1/gems/scss_lint-0.38.0/lib/scss_lint/linter.rb:131:in `visit'
    /var/lib/gems/1.9.1/gems/scss_lint-0.38.0/lib/scss_lint/linter.rb:142:in `block in visit_children'
    /var/lib/gems/1.9.1/gems/scss_lint-0.38.0/lib/scss_lint/linter.rb:140:in `each'
    /var/lib/gems/1.9.1/gems/scss_lint-0.38.0/lib/scss_lint/linter.rb:140:in `visit_children'
    /var/lib/gems/1.9.1/gems/sass-3.4.13/lib/sass/tree/visitors/base.rb:38:in `visit'
    /var/lib/gems/1.9.1/gems/scss_lint-0.38.0/lib/scss_lint/linter.rb:131:in `visit'
    /var/lib/gems/1.9.1/gems/scss_lint-0.38.0/lib/scss_lint/linter.rb:142:in `block in visit_children'
    /var/lib/gems/1.9.1/gems/scss_lint-0.38.0/lib/scss_lint/linter.rb:140:in `each'
    /var/lib/gems/1.9.1/gems/scss_lint-0.38.0/lib/scss_lint/linter.rb:140:in `visit_children'
    /var/lib/gems/1.9.1/gems/sass-3.4.13/lib/sass/tree/visitors/base.rb:38:in `visit'
    /var/lib/gems/1.9.1/gems/scss_lint-0.38.0/lib/scss_lint/linter.rb:131:in `visit'
    /var/lib/gems/1.9.1/gems/scss_lint-0.38.0/lib/scss_lint/linter.rb:142:in `block in visit_children'
    /var/lib/gems/1.9.1/gems/scss_lint-0.38.0/lib/scss_lint/linter.rb:140:in `each'
    /var/lib/gems/1.9.1/gems/scss_lint-0.38.0/lib/scss_lint/linter.rb:140:in `visit_children'
    /var/lib/gems/1.9.1/gems/sass-3.4.13/lib/sass/tree/visitors/base.rb:38:in `visit'
    /var/lib/gems/1.9.1/gems/scss_lint-0.38.0/lib/scss_lint/linter.rb:131:in `visit'
    /var/lib/gems/1.9.1/gems/scss_lint-0.38.0/lib/scss_lint/linter.rb:25:in `run'
    /var/lib/gems/1.9.1/gems/scss_lint-0.38.0/lib/scss_lint/runner.rb:48:in `run_linter'
    /var/lib/gems/1.9.1/gems/scss_lint-0.38.0/lib/scss_lint/runner.rb:29:in `block in find_lints'
    /var/lib/gems/1.9.1/gems/scss_lint-0.38.0/lib/scss_lint/runner.rb:27:in `each'
    /var/lib/gems/1.9.1/gems/scss_lint-0.38.0/lib/scss_lint/runner.rb:27:in `find_lints'
    /var/lib/gems/1.9.1/gems/scss_lint-0.38.0/lib/scss_lint/runner.rb:17:in `block in run'
    /var/lib/gems/1.9.1/gems/scss_lint-0.38.0/lib/scss_lint/runner.rb:16:in `each'
    /var/lib/gems/1.9.1/gems/scss_lint-0.38.0/lib/scss_lint/runner.rb:16:in `run'
    /var/lib/gems/1.9.1/gems/scss_lint-0.38.0/lib/scss_lint/cli.rb:54:in `scan_for_lints'
    /var/lib/gems/1.9.1/gems/scss_lint-0.38.0/lib/scss_lint/cli.rb:48:in `act_on_options'
    /var/lib/gems/1.9.1/gems/scss_lint-0.38.0/lib/scss_lint/cli.rb:27:in `run'
    /var/lib/gems/1.9.1/gems/scss_lint-0.38.0/bin/scss-lint:6:in `<top (required)>'
    /usr/local/bin/scss-lint:23:in `load'
    /usr/local/bin/scss-lint:23:in `<main>'

**Comment by bassjobsen** [comment]
seems to be due to  accidentally using ` test #{a}` instead of  test `#{$a}`.

**Comment by sds** [comment]
Hey @bassjobsen, thanks for the report. 

Even though your issue was fixed by adding a `$`, this is still a crash on what could be argued is valid input.

I've implemented a fix in a9b8fe3963265af1e41fc7417801640e53564921.
